presto-hive/src/main/java/com/facebook/presto/hive/WriteCompletedEvent.java

```diff
@@ -0,0 +1,127 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.hive;
+
+import io.airlift.event.client.EventField;
+import io.airlift.event.client.EventType;
+
+import javax.annotation.concurrent.Immutable;
+
+import static java.lang.String.format;
+
+@Immutable
+@EventType("WriteCompletedEvent")
+public class WriteCompletedEvent
+{
+    private final String queryId;
+    private final String fileSystemUri;
+    private final String schemaName;
+    private final String tableName;
+    private final String partitionName;
+    private final String storageFormat;
+    private final String writerImplementation;
+    private final String prestoVersion;
+    private final long bytes;
+    private final long rows;
+
+    public WriteCompletedEvent(
+            String queryId,
+            String fileSystemUri,
+            String schemaName,
+            String tableName,
+            String partitionName,
+            String storageFormat,
+            String writerImplementation,
+            String prestoVersion,
+            long bytes,
+            long rows)
+    {
+        this.queryId = queryId;
+        this.fileSystemUri = fileSystemUri;
+        this.schemaName = schemaName;
+        this.tableName = tableName;
+        this.partitionName = partitionName;
+        this.storageFormat = storageFormat;
+        this.writerImplementation = writerImplementation;
+        this.prestoVersion = prestoVersion;
+        this.bytes = bytes;
+        this.rows = rows;
+    }
+
+    @EventField
+    public String getQueryId()
+    {
+        return queryId;
+    }
+
+    @EventField
+    public String getFileSystemUri()
+    {
+        return fileSystemUri;
+    }
+
+    @EventField
+    public String getSchemaName()
+    {
+        return schemaName;
+    }
+
+    @EventField
+    public String getTableName()
+    {
+        return tableName;
+    }
+
+    @EventField
+    public String getPartitionName()
+    {
+        return partitionName;
+    }
+
+    @EventField
+    public String getStorageFormat()
+    {
+        return storageFormat;
+    }
+
+    @EventField
+    public String getWriterImplementation()
+    {
+        return writerImplementation;
+    }
+
+    @EventField
+    public String getPrestoVersion()
+    {
+        return prestoVersion;
+    }
+
+    @EventField
+    public long getBytes()
+    {
+        return bytes;
+    }
+
+    @EventField
+    public long getRows()
+    {
+        return rows;
+    }
+
+    public String toString()
+    {
+        return format("Query %s finishes write URI: %s, schema: %s, table: %s, partition: %s.",
```

Comment:
This is not the correct formatting for a `toString`. Move this to the logging statement and remove this method.

Also, don't end in a period, since it gets mixed with the value